src/status_im2/contexts/quo_preview/share/share_qr_code.cljs

```diff
@@ -1,34 +1,146 @@
+                             :customization-color :purple
+                             :emoji               "🐈"
+                             :on-info-press       #(js/alert "Info pressed")
+                             :on-legacy-press     #(js/alert (str "Tab " % " pressed"))
+                             :on-multichain-press #(js/alert (str "Tab " % " pressed"))
+                             :networks            (take 2 possible-networks)
+                             :on-settings-press   #(js/alert "Settings pressed")})]
     (fn []
-      (let [qr-media-server-uri (image-server/get-qr-image-uri-for-any-url
-                                 {:url         (:url @state)
+      (let [qr-url              (if (= (:type @state) :wallet-multichain)
+                                  (as-> (:networks @state) $
+                                    (map get-network-short-name-url $)
+                                    (apply str $)
+                                    (str $ (:qr-data @state)))
+                                  (:qr-data @state))
+            qr-media-server-uri (image-server/get-qr-image-uri-for-any-url
+                                 {:url         qr-url
                                   :port        (rf/sub [:mediaserver/port])
-                                  :qr-size     300
-                                  :error-level :highest})]
+                                  :qr-size     500
+                                  :error-level :highest})
+            typed-descriptor    (concat descriptor
+                                        (case (:type @state)
+                                          :profile           profile-descriptor
+                                          :wallet-legacy     wallet-legacy-descriptor
+                                          :wallet-multichain wallet-multichain-descriptor
+                                          nil))]
         [preview/preview-container
          {:state                     state
-          :descriptor                descriptor
-          :component-container-style {:padding-vertical 20}}
-         [quo/share-qr-code
-          {:qr-image-uri      qr-media-server-uri
-           :link-title        (:link-title @state)
-           :url-on-press      #(js/alert "url pressed")
-           :url-on-long-press #(js/alert "url long pressed")
-           :share-on-press    #(js/alert "share pressed")
-           :qr-url            (:url @state)}]]))))
+          :descriptor                typed-descriptor
+          :component-container-style {:padding-horizontal 0}}
+         [rn/view
+          {:style {:flex               1
+                   :justify-content    :flex-end
+                   :align-items        :center
+                   :padding-horizontal 20
+                   :padding-vertical   30}}
+          ;; Hack to reset the `:qr-data` atom value when the `:type` changes
+          [:f> reset-qr-data (:type @state) state]
```

Comment:
@ilmotta 

When the `type` (:profile, :wallet-legacy or :wallet-multichain) of the selected component changes, I should also change the `:qr-data`  value, from being a profile link to be a wallet address.

Sadly, we don't have that capability in our preview infrastrcture, so I did a hack to achieve that behaviour. The proper way would be something like this:

```clojure
[preview/preview-container
 {:state      state
  :descriptor descriptor
  :on-change  (fn [{new-type :type :as new-state}]
                (if (= new-type :a)
                  (swap! state assoc :label "Showing a")
                  (swap! state assoc :label "Showing something else")))}
 [rn/view
 ,,,]]
 
 WDYT?
```

Reply:
Thanks @ilmotta 
really interesting approach to solve it! :)